src/main/java/com/Kuba2412/medicalclinic/service/PatientService.java

```diff
@@ -2,52 +2,45 @@
 
 import com.Kuba2412.medicalclinic.model.mapper.PatientMapper;
 import com.Kuba2412.medicalclinic.model.dto.PatientDTO;
-import com.Kuba2412.medicalclinic.model.User;
 import com.Kuba2412.medicalclinic.repository.PatientRepository;
+import jakarta.transaction.Transactional;
 import lombok.RequiredArgsConstructor;
 import com.Kuba2412.medicalclinic.model.Patient;
 
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
-
 @Service
 @RequiredArgsConstructor
 public class PatientService {
 
     private final PatientRepository patientRepository;
-    private final UserService userService;
     private final PatientMapper patientMapper;
 
     public PatientDTO getPatientDtoByEmail(String email) {
         Patient patient = patientRepository.findByEmail(email)
                 .orElseThrow(() -> new IllegalArgumentException("Patient not found"));
         return patientMapper.patientToPatientDTO(patient);
     }
-    public Patient getPatientByEmail(String email) {
-        return patientRepository.findByEmail(email)
-                .orElseThrow(() -> new IllegalArgumentException("Patient not found"));
-    }
 
-    public List<PatientDTO> getPatientDtosByFirstName(String firstName) {
-        List<Patient> patients;
+    @Transactional
+    public List<PatientDTO> getPatientDtosByFirstName(String firstName, Pageable pageable) {
+        Page<Patient> patientsPage;
         if (firstName != null) {
```

Comment:
mozna to bylo napisac ladniej w jednej linicje:
List<Patient> patients = firstName != null  ? patientRepository.findByFirstName(firstName, pageable).getContent() : patientRepository.findAll(pageable).getContent(); 

Nie mamy calego ifa dzieki wykorzystaniu operatora trojargumentowego